core/render/vi/__init__.py

```diff
@@ -77,56 +67,45 @@ def getStructure(adminTree, module):
     return json.dumps(res or None, cls=CustomJsonEncoder)
 
 
+@exposed
 def setLanguage(lang, skey):
     if not securitykey.validate(skey):
         return
     if lang in conf["viur.availableLanguages"]:
         currentLanguage.set(lang)
 
 
-setLanguage.exposed = True
-
-
-def dumpConfig(adminTree):
-    adminConfig = {}
-    for key in dir(adminTree):
-        app = getattr(adminTree, key)
-        if "adminInfo" in dir(app) and app.adminInfo:
-            if callable(app.adminInfo):
-                info = app.adminInfo()
-                if info is not None:
-                    adminConfig[key] = info
-            else:
-                adminConfig[key] = app.adminInfo.copy()
-                adminConfig[key]["name"] = str(adminConfig[key]["name"])
-                adminConfig[key]["views"] = []
-                if "views" in app.adminInfo:
-                    for v in app.adminInfo["views"]:
-                        tmp = v.copy()
-                        tmp["name"] = str(tmp["name"])
-                        adminConfig[key]["views"].append(tmp)
-        if key in adminConfig:
-            adminConfig[key]["indexes"] = getattr(app, "indexes", [])
+@exposed
+def dumpConfig():
+    res = {}
+
+    for key in dir(conf["viur.mainApp"].vi):
+        module = getattr(conf["viur.mainApp"].vi, key, None)
+        if not isinstance(module, Module):
+            continue
+
+        if admin_info := module.describe():
+            res[key] = admin_info
 
     res = {
-        "modules": adminConfig,
-        "configuration": {}
+        "modules": res,
+        "configuration": {k.removeprefix("admin."): v for k, v in conf.items() if k.lower().startswith("admin.")}
     }
 
-    for k, v in conf.items():
-        if k.lower().startswith("admin."):
-            res["configuration"][k[6:]] = v
-
     currentRequest.get().response.headers["Content-Type"] = "application/json"
     return json.dumps(res)
 
 
+@exposed
 def getVersion(*args, **kwargs):
-    # We force the patch-level of our version to be always zero for security reasons
-    return json.dumps((conf["viur.version"][0], conf["viur.version"][1], 0))
-
+    """
+    Returns viur-core version number
+    """
+    if conf["viur.instance.is_dev_server"]:
+        json.dumps(conf["viur.version"])
```

Comment:
```suggestion
    if conf["viur.instance.is_dev_server"]:
        return json.dumps(conf["viur.version"])
```
We should return it ?